AS canary 14 mods

stevemessick opened the pull request:
Here are the changes required to get our plugin to build with Android Studio canary 14.

I still have some debugging to do to see exactly why hot reload is broken.

@devoncarew @pq

stevemessick commented:
This is now ready for review. Got a clean build on travis and it builds locally, in both dev and release modes. It installs into Android Studio and IntelliJ although I have not tested it (and suspect the hot reload issue I observed earlier is still there). 

The changes to the .iml files do not affect travis but will cause problems for development using old versions of IntelliJ.

pq commented on .travis.yml:
```diff
@@ -23,5 +23,5 @@ script: ./tool/travis.sh
-  - IDEA_VERSION=2017.3
-  - IDEA_VERSION=2018.1
+  - CI=true IDEA_VERSION=2017.3
+  - CI=true IDEA_VERSION=2018.1
```
Curious what `CI=true` does?

stevemessick replied:
CI identifies the run as a travis CI run to the plugin tool. It is used during the build to auto-edit some files on travis, so no one has to hack the plugin.dart code to make a local build work.

pq replied:
Cool.  Thanks for the info!

pq commented on flutter-intellij-community.iml:
```diff
@@ -42,7 +42,25 @@
+    <orderEntry type="module-library">
+      <library>
+        <CLASSES>
+          <root url="jar://$APPLICATION_HOME_DIR$/lib/miglayout-swing-5.0.jar!/" />
```
This looks a little brittle to me.  If we're going to depend on a specific jar version would it be safer to put it in a deps dir like was done for `weberknecht-0.1.5.jar`?

stevemessick replied:
That would mean distributing the jar with the plugin rather than using the jar distributed with the IDE. @scheglov How was this dependency defined prior to 2018.3?

pq replied:
@devoncarew may have ideas but it was just an observation. I'm fine with whatever.

devoncarew replied:
> @devoncarew may have ideas but it was just an observation. I'm fine with whatever.

I'd want to make sure that this import works for AS 3.1 / 3.2  and IntelliJ 2017.3 and 2018.1, on the bot and for users in the wild.

pq commented on tool/plugin/lib/plugin.dart:
```diff
@@ -530,24 +530,44 @@ class BuildCommand extends ProductCommand {
+            newSource.replaceAll('IconUtil.toImage', ''); // Leaves parens
+        processedFile1.writeAsStringSync(newSource);
+        processedFile2 = new File(
+            'flutter-studio/src/io/flutter/project/ChoseProjectTypeStep.java');
```
Aside: should this file get renamed to `ChooseProjectTypeStep`?

stevemessick replied:
Yes, but fixing bad file names is outside the scope of this PR :)

pq replied:
😃

stevemessick commented:
Got the weird problems sorted out. Fixed a compile issue that caused an exception in Canary 15. The build still uses Canary 14. Deleted the mig-layout dependency and the CI parameter. Also fixed a problem that caused the build to use the wrong source dependencies in some situations.

PTAL, I think this one is ready to go.

devoncarew commented:
> Also fixed a problem that caused the build to use the wrong source dependencies in some situations.

awesome!

devoncarew commented:
@stevemessick, looks like travis is failing on a unneeded `@override`: https://travis-ci.org/flutter/flutter-intellij/jobs/381881343#L943

devoncarew commented on flutter-intellij-community.iml:
```diff
@@ -42,7 +42,7 @@
     <orderEntry type="library" name="snakeyaml" level="project" />
-    <orderEntry type="library" name="Dart SDK" level="project" />
-    <orderEntry type="library" name="Dart Packages" level="project" />
+    <orderEntry type="library" exported="" name="Dart SDK" level="project" />
```
Is the `exported=""` necessary here? likely harmless w/ or w/o it?

stevemessick replied:
No idea. I didn't edit this file by hand.

devoncarew commented on tool/plugin/lib/plugin.dart:
```diff
@@ -490,7 +490,7 @@ class BuildCommand extends ProductCommand {
       }
       if (!await performReleaseChecks(this)) {
-        return 1;
+        //return 1;
```
This should still return `1` (to fail)?

stevemessick replied:
Good catch. Too much git commanding. Fixed.

stevemessick commented:
That's fixed now, @devoncarew . The override is only valid for Canary 15.

devoncarew commented on tool/plugin/lib/plugin.dart:
```diff
@@ -530,24 +531,46 @@ class BuildCommand extends ProductCommand {
-      var processedFile = new File(
-          'flutter-studio/src/io/flutter/module/FlutterDescriptionProvider.java');
-      var oldSource;
+      var processedFile1, processedFile2, oldSource1, oldSource2, newSource;
```
If we end up with more of these, we'll likely want to wrap in a class, and have something along the lines off a list of them that we can `appy()` and `revert()`.
